CONTRIBUTING.md

```diff
@@ -0,0 +1,187 @@
+# Contributing
+
+Thanks for taking a look at StatsBase!
+
+In general this guide summarizes and extends the [Julia Contributing Guidelines](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#general-formatting-guidelines-for-julia-code-contributions) and [Julia Style Guide](https://docs.julialang.org/en/latest/manual/style-guide.html).
+
+## Reporting Issues
+
+* It's always helpful to start by reviewing existing issues is and if necessary referencing back to previous discussions.
+* Including minimal sample code (either to reproduce a bug or demonstrating desired behaviour) is greatly appreciated.
+* For bugs, it's helpful to checkout the latest development version (`Pkg.checkout("StatsBase")`) and post the output from `versioninfo()`.
+
+## Pull Requests (PR)s
+
+* Try to keep changes in PRs constrained if possible. This will reduce the amount of work for the reviewer and likely speed up the review process.
+* Using `git diff upstream/master` and `git diff --stat upstream/master` (assuming you've run `git remote add upstream https://github.com/JuliaStats/StatsBase.jl`) can be useful commands for comparing your branch against the upstream master before pushing.
+* Running tests locally with `Pkg.test("StatsBase")` before committing can be a helpful way of reducing noise.
+* [Coverage](https://github.com/JuliaCI/Coverage.jl) can be useful for checking if you've lowered the overall test coverage against [master](https://coveralls.io/r/JuliaStats/StatsBase.jl?branch=master)
+
+## Style
+
+### Summary
+
+* Use 4 spaces per indentation level, no tabs.
```

Comment:
I agree regarding reiterating certain concrete points such as whitespace. (Partially because I care way too much about indentation.) I bet a significant number of contributors won't be bothered to follow a link, so it's a good idea to make sure they at least see a few key points.